dev-guide/kubernetes-continuous-rebase.md

```diff
@@ -0,0 +1,153 @@
+---
+title: kubernetes-rebase
+authors:
+  - "@fbertina"
+reviewers:
+  - "@soltysh"
+approvers:
+  - "@soltysh"
+creation-date: 2023-06-15
+last-updated: 2023-06-15
+---
+
+# Kubernetes Continuous Rebase
+
+## Goal
+
+The main goal of this proposal is to proactively identify and address
+any potential issues that may arise during the upcoming rebase
+process.
+
+The desired outcome is to be able to land the rebase PR significantly
+earlier in the process, potentially aligning with the release of the
+upstream tag.
+
+## Proposal
+
+Currently, the rebase work is typically spread out over a period of 1
+or 2 months. However, it can potentially be distributed throughout the
+development cycle of Kubernetes. To achieve this, we could have an OCP
+branch with a continually updated Kubernetes codebase, allowing most
+of the work to be completed even before the rebase process begins.
+
+The main approach involves applying our downstream patches against the
+upstream master branch on a daily basis. It is expected that some
+patches may fail to be applied multiple times during the development
+cycle. However, as soon as such failures occur, we will receive
+notifications, and the necessary fixes will be applied to the
+downstream patches or the upstream code.
+
+Implementing this approach brings several benefits:
+
+1. The rebase process becomes less time-sensitive.
+2. We receive early signals if an upstream change breaks OCP, enabling
+   us to address the issue promptly either in the upstream code or on
+   our side.
+3. The rebase PR should be ready to be landed as soon as the upstream
+   code becomes generally available (GA).
+
+To implement this proposal, the following steps are required:
+
+### Watcher
+
+For each OCP release, we will designate a watcher to participate in
+the process. Ideally, it should be the same person who will execute
+the final rebase.
+
+A watcher is responsible for ensuring that the remaining steps
+outlined below are executed without errors.
+
+Although some manual work is required, it should not occupy their
+entire daily working time.
+
+### A -next branch
+
+For each of the dependencies listed below, a new branch called
+`ocp-next` is created with their Kubernetes dependencies updated:
+
+* openshift/api
+* openshift/client-go
+* openshift/library-go
+* openshift/apiserver-library-go
```

Comment:
I'm not sure this is needed, so I'd mark this bit as optional. Based on my experience from the past several rebases, I was able to get away w/o these changes.

Reply:
Right, this should be optional.